### PR TITLE
Convert logging f-strings to use % format, part four

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98765
* __->__ #98705
* #98704

This does multi-line concatenated string literals.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by voznesenskym — APPROVED** [review]

File: torch/distributed/elastic/multiprocessing/errors/error_handler.py [hunk 1]
```diff
@@ -114,8 +116,9 @@ def dump_error_file(self, rootcause_error_file: str, error_code: int = 0):
             if error_code:
                 self.override_error_code_in_rootcause_data(rootcause_error_file, rootcause_error, error_code)
             log.debug(
-                f"child error file ({rootcause_error_file}) contents:\n"
-                f"{json.dumps(rootcause_error, indent=2)}"
+                "child error file (%s) contents:\n"
+                "%s",
+                rootcause_error_file, json.dumps(rootcause_error, indent=2)
```
Comment (voznesenskym) [inline torch/distributed/elastic/multiprocessing/errors/error_handler.py:121]: I would double check that %s does the right thing with json, but I think it should?

File: torch/distributed/launcher/api.py [hunk 2]
```diff
@@ -188,19 +188,23 @@ def launch_agent(
...
+        "  rdzv_endpoint    : %s\n"
+        "  rdzv_configs     : %s\n"
+        "  max_restarts     : %s\n"
+        "  monitor_interval : %s\n"
+        "  log_dir          : %s\n"
+        "  metrics_cfg      : %s\n",
+        entrypoint_name, config.min_nodes, config.max_nodes,
+        config.nproc_per_node, config.run_id, config.rdzv_backend,
+        config.rdzv_endpoint, config.rdzv_configs, config.max_restarts,
+        config.monitor_interval, config.log_dir, config.metrics_cfg
```
Comment (voznesenskym) [inline torch/distributed/launcher/api.py:207]: this is a lot worse imo

File: torch/distributed/elastic/multiprocessing/errors/error_handler.py (hunk 1 above)
Comment (ezyang) [reply torch/distributed/elastic/multiprocessing/errors/error_handler.py:121]: it will implicitly call str which is what the old code did

**Pull request closed without merging**